addons/stock/models/stock_picking.py

```diff
@@ -729,6 +729,44 @@ def _compute_return_count(self):
         for picking in self:
             picking.return_count = len(picking.return_ids)
 
+    def _search_products_availability_state(self, operator, value):
+        def get_stock_moves(moves, state):
```

Comment:
Maybe it could be better to make this a `stock.move` method to avoid duplication (almost exact same function in `_search_components_availability_state`.)
You can pass the date as a parameter for example